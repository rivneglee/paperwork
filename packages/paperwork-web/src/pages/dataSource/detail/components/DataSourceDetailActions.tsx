import React, { FunctionComponent } from 'react';
import { Button, ButtonRow, Icons } from '@paperwork/ui-widgets';

interface Props {
  isCreating: boolean;
  onClickSave: () => void;
  onClickCancel: () => void;
  onClickDelete: () => void;
}

const DataSourceDetailActions: FunctionComponent<Props> = ({
  isCreating,
  onClickDelete,
  onClickCancel,
  onClickSave,
}) => {
  const secondaryButtons = isCreating ? [] : [
    <Button key="delete" onClick={onClickDelete} size="m" color="danger" icon={<Icons.Delete />}>Delete</Button>,
  ];

  return (
    <ButtonRow
      primary={[
        <Button
          key="save"
          color="primary"
          size="m"
          icon={<Icons.Save />}
          onClick={onClickSave}
        >
          Save
        </Button>,
        <Button key="cancel" onClick={onClickCancel} size="m" icon={<Icons.Cancel/>}>Cancel</Button>,
      ]}
      secondary={secondaryButtons}
    />
  );
};

export default DataSourceDetailActions;
