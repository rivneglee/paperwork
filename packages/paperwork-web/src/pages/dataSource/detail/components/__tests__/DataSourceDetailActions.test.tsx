import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import DataSourceDetailActions from '../DataSourceDetailActions';

describe('DataSourceDetailActions', () => {
  const commonProps = {
    onClickDelete: jest.fn(),
    onClickCancel: jest.fn(),
    onClickSave: jest.fn(),
  };

  it('should render properly when creating new', () => {
    const wrapper = shallow(
      <DataSourceDetailActions {...commonProps} isOwner={true} isCreating={true} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly when editing collaborative datasource', () => {
    const wrapper = shallow(
      <DataSourceDetailActions {...commonProps} isOwner={false} isCreating={false} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly when editing owned datasource', () => {
    const wrapper = shallow(
      <DataSourceDetailActions {...commonProps} isOwner={true} isCreating={false} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
