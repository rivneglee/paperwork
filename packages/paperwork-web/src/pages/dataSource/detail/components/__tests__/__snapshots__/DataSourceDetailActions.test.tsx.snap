// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSourceDetailActions should render properly when creating new 1`] = `
<Ei
  primary={
    Array [
      <si
        color="primary"
        icon={<Save />}
        onClick={[Function]}
        size="m"
    >
        Save
    </si>,
      <si
        icon={<Cancel />}
        onClick={[Function]}
        size="m"
    >
        Cancel
    </si>,
    ]
  }
  secondary={Array []}
/>
`;

exports[`DataSourceDetailActions should render properly when editing collaborative datasource 1`] = `
<Ei
  primary={
    Array [
      <si
        icon={<Cancel />}
        onClick={[Function]}
        size="m"
    >
        Cancel
    </si>,
    ]
  }
  secondary={Array []}
/>
`;

exports[`DataSourceDetailActions should render properly when editing owned datasource 1`] = `
<Ei
  primary={
    Array [
      <si
        color="primary"
        icon={<Save />}
        onClick={[Function]}
        size="m"
    >
        Save
    </si>,
      <si
        icon={<Cancel />}
        onClick={[Function]}
        size="m"
    >
        Cancel
    </si>,
    ]
  }
  secondary={
    Array [
      <si
        color="danger"
        icon={<Delete />}
        onClick={[Function]}
        size="m"
    >
        Delete
    </si>,
    ]
  }
/>
`;
